src/components/Board/Board.tsx

```diff
@@ -23,31 +26,54 @@ interface IBoardProps {
 export const Board = React.memo(
   function Board({ cardLists }: IBoardProps) {
     const [isFormShown, setFormShown] = React.useState(false);
+    const [activeDraggableId, setActiveDraggableId] = React.useState<string | null>(null);
     const [formValue, setFormValue] = React.useState<string>('');
-    const cardsByListId = useSelector(selectCardsByListId) || [];
+    const cardsByListId = useSelector(selectCardsByListId);
+    const cardsById = useSelector(selectCardsById);
     const dispatch = useDispatch();
     const sensors = useSensors(useSensor(MouseSensor), useSensor(TouchSensor));
 
-    function handleDragEnd({ active, over }: DragEndEvent) {
-      if (over === null) {
+    const handleDragStart = ({ active }: DragStartEvent) => {
+      setActiveDraggableId(String(active.id));
+    };
+
+    const handleDragEnd = ({ active, over }: DragEndEvent) => {
+      const activeType = active.data.current?.type;
+
+      if (over === null || active.id === over.id) {
         return;
       }
-      const id = String(active.id);
-      const targetIndex = over.data.current ? over.data.current.index : 0;
 
-      dispatch(
-        moveCard({
-          id,
-          targetIndex,
-        })
-      );
-    }
+      const activeId = String(active.id);
+      const overId = String(over.id);
+
+      if (activeType === 'item') {
```

Comment:
This check is not required as far as we do not have any other items that could be reordered (e.g. lists)